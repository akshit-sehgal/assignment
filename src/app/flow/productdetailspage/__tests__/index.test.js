import React from 'react';
import { shallow } from 'enzyme';
import ProductDetailsPage from '../';

describe('Testing <ProductDetailsPage />', () => {
    let props;
    beforeEach(() => {
        props = {
            history: {
                push: jest.fn()
            },
            match: {
                params: {
                    id: '1'
                }
            }
        }
    })
    it('should render the default view', () => {
        const wrapper = shallow(<ProductDetailsPage {...props} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('Connect(Header)').length).toEqual(1);
        expect(wrapper.find('Connect(ProductDetails)').length).toEqual(1);
        expect(wrapper.find('Footer').length).toEqual(1);
    });

});