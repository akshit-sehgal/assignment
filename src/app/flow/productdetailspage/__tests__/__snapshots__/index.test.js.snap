// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <ProductDetailsPage /> should render the default view 1`] = `
<div
  className="products-details-container"
>
  <Connect(Header)
    navigate={[Function]}
  />
  <Connect(ProductDetails)
    id="1"
  />
  <Footer />
</div>
`;
